web/src/views/InvestigationDetail.vue

```diff
@@ -498,17 +505,84 @@ export default defineComponent({
       }));
     }
 
+    async function toggleNonTiledImageOverlay(pin: Pin, dataset: Dataset) {
+      if (!dataset.content) return;
+      if (!featureLayer) return; // sanity check. this should exist
+      if (!nonTiledOverlayFeature) {
+        nonTiledOverlayFeature = featureLayer.createFeature('quad');
+      }
+      const quadData = nonTiledOverlayFeature.data() || [];
+      // eslint-disable-next-line
+      const existingOverlay = quadData.find((overlay: any) => overlay.pinId === pin.id);
+      if (existingOverlay) {
+        nonTiledOverlayFeature.data(
+          // eslint-disable-next-line
+          nonTiledOverlayFeature.data().filter((overlay: any) => overlay.pinId !== pin.id),
+        ).draw();
+        nonTiledOverlayFeature.draw();
+      } else {
+        try {
+          const urlRoot = process.env.VUE_APP_API_ROOT;
+          const url = `${urlRoot}/datasets/tile_source/${dataset.id}/thumbnail.png`;
+          const image: HTMLImageElement = await getNonTiledImage(url);
+          image.crossOrigin = 'use-credentials';
+          if (store.state.axiosInstance) {
+            const imageMetadata = await (await store.state.axiosInstance.get(`${urlRoot}/datasets/tile_source/${dataset.id}/metadata`)).data;
+            const ul = postGisToPoint(pin.child_location) || { x: 0, y: 0 };
+            const lr = {
+              x: ul.x + (imageMetadata.sizeX || 0),
+              y: ul.y + (imageMetadata.sizeY || 0),
+            };
+            quadData.push({
+              ul,
+              lr,
+              image,
+              pinId: pin.id,
+            });
+            nonTiledOverlayFeature.data(quadData).draw();
+          }
+        } catch (error) {
+          // eslint-disable-next-line
+          console.error(`Failed to show overlay for dataset ${dataset.id}.`);
```

Comment:
You can throw an `Error` object to avoid the eslint disabling line.

Reply:
In that case would it be better to remove the try/catch block? I don't know if re-throwing with a custom error message will be meaningful at this point. I've removed the try/catch 1fe6e72; I'm interested to head your opinion on that change.

Reply:
The question is, what is the try/catch block doing there specifically?

According to the [`await` docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await) the value a promise is rejected with will be thrown. So, to be typesafe I think it's best to wrap the `await` expression in a try/catch in order to rethrow the value that arose, or other handle it. You could put the try/catch around a longer block of code but I think that choice depends on how you want to route control flow in case of a rejection promise.

Let me know if this makes sense--if so, I can help narrow down a design for where try/catch should be in this code.

Reply:
[Here](https://axios-http.com/docs/handling_errors) is the relevant axios documentation. I've added a `.catch()` to the call to `toggleNonTiledImageOverlay`. If this function throws an error due to a response outside the 2XX range, it is because the associated dataset either has no content or its content couldn't be opened with `django-large-image`. An error is thrown with a specific message in this case. If an error is thrown due to no response or an inability to make the request, I simply re-throw that error. This change was made in dcc17c7